wrapper/pre_copy.py

```diff
@@ -698,6 +715,36 @@ def _get_nbdkit_cmd(self, disk, filters, final):
 
         return nbdkit_cmd
 
+    def _get_nbdkit_cmd_ssh(self, disk, filters, final):
+        print("*** %s" % disk.absolute_path.replace('.vmdk', '-flat.vmdk'))
+        nbdkit_cmd = [
+            'nbdkit',
+            '-v',
+            '-U', disk.sock,
+            '-P', disk.pidfile,
+            '--exit-with-parent',
+            '--readonly',
+            '--foreground',
+            '--exportname=/',
+        ] + [
+            '--filter=' + f for f in filters
+        ] + [
+            '--filter=log',
+            'ssh',
+            # pylint: disable=protected-access
+            'host=%s' % self.vmware.server,
+            'password=+%s' % self._vmware_password_file,
+            'path=%s' % disk.absolute_path.replace('.vmdk', '-flat.vmdk'),
```

Comment:
Something like this would be safer:
```suggestion
            'path=%s' % '-flat'.join(os.path.splitext(disk.absolute_path.replace()),
```

Reply:
Or, if there might be other files than just `.vmdk`:
```
root, ext = os.path.splitext(disk.absolute_path)
if ext == 'vmdk':
    root = root + '-flat'
path = root + '.' + ext
```

Reply:
You can also doit right away when assigning into the absolute path, so that here it is really just constructing a command line, not doing other logic.

Reply:
That should always be a `.vmdk` file. And the logic here is to derive the flat file name from the absolute path, as the absolute path point to the generic vmdk file.